### PR TITLE
[IMP] stock and mrp : Usability improvements

### PR DESCRIPTION
**Pull request by hdh-odoo**
**Description of the issue/feature this PR addresses**: 
- Task: https://www.odoo.com/web#id=1943397&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720
- Pad: https://pad.odoo.com/p/r.018c7fbae8a5a0757b9b1cc5ce2b1d01

**Description:**
        - Removed "Assembly station 1" from everywhere and replaced it with "Assemmbly Line 1". Renamed "Assembly station 2" to "Assembly Line 2".
	- operations field on BOM will be hidden when advanced routes are not active.
	- Hide sequence field in form view in BOM and added a widget handle in list view.
	- Hide field "Manufacturing readiness" in MO when WO aren't active.
	- Added a domain in PLM in Lot/Serial Numbers to show only variants of the product.
	- Removed red and green colors from Traceability Report.
	- Renamed "Cost" to "Value" on products.
	- Changed tooltip of "weight for shipping" in shipping information in inventory.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/views/stock_production_lot_views.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
                 </div>
                 <group name="main_group">
                     <group>
-                        <field name="product_id" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."/>
+                        <field name="product_id" domain="[('id', context.get('operator', '!='), context.get('products', False))]" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved." />
```
Comment (sle-odoo) [inline addons/stock/views/stock_production_lot_views.xml]: this is wrong

File: addons/stock/models/product.py [hunk 2]
```diff
@@ -670,11 +670,12 @@ def action_open_product_lot(self):
         self.ensure_one()
         action = self.env.ref('stock.action_production_lot_form').read()[0]
         action['domain'] = [('product_id.product_tmpl_id', '=', self.id)]
+        action['context'] = {'products': self.product_variant_ids.ids, 'operator': 'in'}
```
[the remark this answers is not in the document]
Comment (hsh-odoo) [reply addons/stock/models/product.py]: @sle we have passed dynamic domain because we want to show only same product when we open from the product.product but when we open lot view from product template we have to show all the products of template.

Now action domain will apply on the lot record not on the product domain.
we can't set static domain because operator will be change.

we have tried to put full domain in the context but it is not working.

if you have better solution then please provide us.

[the remark this answers is not in the document]
Comment (sle-odoo) [reply addons/stock/models/product.py]: operator will be change? why?

[the remark this answers is not in the document]
Comment (hsh-odoo) [reply addons/stock/models/product.py]: @sle-odoo 

When we open from the template we need to show all the variants 
so domain will be ('id', 'in', variants) but when form open product ( variant ) it should be 
('id', '=', variant) ... to not show other variant 

is this possible by just domain change ?

File: addons/stock/models/product.py [hunk 3]
```diff
@@ -403,8 +403,7 @@ def action_view_related_putaway_rules(self):
     def action_open_product_lot(self):
         self.ensure_one()
         action = self.env.ref('stock.action_production_lot_form').read()[0]
-        action['domain'] = [('product_id', '=', self.id)]
```
Comment (hmo-odoo) [inline addons/stock/models/product.py]: @hdh-odoo , do not remove domain

File: addons/stock/models/product.py [hunk 4]
```diff
@@ -669,12 +668,11 @@ def action_view_stock_move_lines(self):
     def action_open_product_lot(self):
         self.ensure_one()
         action = self.env.ref('stock.action_production_lot_form').read()[0]
-        action['domain'] = [('product_id.product_tmpl_id', '=', self.id)]
```
Comment (hmo-odoo) [inline addons/stock/models/product.py]: @hdh-odoo , do not remove domain

File: addons/stock/models/product.py (hunk 3 above)
Comment (hdh-odoo) [reply addons/stock/models/product.py]: yes, changes are done.

File: addons/mrp/views/mrp_bom_views.xml [hunk 5]
```diff
@@ -90,12 +90,12 @@
                         <page string="Miscellaneous">
                             <group>
                                 <group>
-                                    <field name="sequence"/>
+                                    <field name="sequence" invisible="1"/>
```
Comment (arbaes) [inline addons/mrp/views/mrp_bom_views.xml]: Why not simply removing the sequence from this here ?

File: addons/stock/models/stock_production_lot.py [hunk 6]
```diff
@@ -63,3 +64,8 @@ def action_lot_open_quants(self):
         action = self.env.ref('stock.lot_open_quants').read()[0]
         action['context'] = {'search_default_lot_id': self.id}
         return action
+
+    @api.onchange('product_tmpl_id')
+    def onchange_product_tmpl_id(self):
+        if self.product_tmpl_id:
+            return {'domain':{'product_id': [('product_tmpl_id', '=', self.product_tmpl_id.id)]}}
```
Comment (arbaes) [inline addons/stock/models/stock_production_lot.py]: Domain will be reset if you edit then save the form by using an onchange to set the domain

File: addons/stock/views/report_stock_traceability.xml [hunk 7]
```diff
@@ -41,7 +38,7 @@
                             </span>
                         </t>
                         <t t-else="">
-                            <span t-att-style="span_color">
+                            <span>
```
Comment (arbaes) [inline addons/stock/views/report_stock_traceability.xml]: Remove span, unless it has a purpose (scss styling maybe ?) there.

File: addons/stock/views/stock_production_lot_views.xml [hunk 8]
```diff
@@ -25,7 +25,8 @@
                 </div>
                 <group name="main_group">
                     <group>
-                        <field name="product_id" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."/>
+                        <field name="product_tmpl_id" invisible="1"/>
+                        <field name="product_id" readonly="context.get('set_product_readonly', False)" force_save="1" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."/>
```
Comment (arbaes) [inline addons/stock/views/stock_production_lot_views.xml]: I would simply check if `default_product_id `is in the context instead

File: addons/stock_account/models/product.py [hunk 9]
```diff
@@ -52,7 +52,7 @@ class ProductProduct(models.Model):
 
     stock_value_currency_id = fields.Many2one('res.currency', compute='_compute_stock_value_currency')
     stock_value = fields.Float(
-        'Value', compute='_compute_stock_value')
+        'Stock Value', compute='_compute_stock_value')
```
Comment (arbaes) [inline addons/stock_account/models/product.py]: I asked @lap-odoo about this. You should rename it to 'Total Value' to keep a correct meaning

File: addons/stock/models/stock_production_lot.py [hunk 10]
```diff
@@ -74,3 +78,8 @@ def action_lot_open_quants(self):
         if self.user_has_groups('stock.group_stock_manager'):
             self = self.with_context(inventory_mode=True)
         return self.env['stock.quant']._get_quants_action()
+
+    @api.onchange('related_product_id')
+    def onchange_related_product_id(self):
+        if self.related_product_id:
+            self.product_id = self.related_product_id
```
Comment (svs-odoo) [inline addons/stock/models/stock_production_lot.py]: So, remove these lines...

File: addons/stock/views/stock_production_lot_views.xml [hunk 11]
```diff
@@ -21,9 +21,9 @@
                 </div>
                 <group name="main_group">
                     <group>
-                        <field name="product_id"
-                               help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."
-                               attrs="{'readonly': [('id', '!=', False), ('product_id', '!=', False)]}"/>
+                        <field name="product_tmpl_id" invisible="1"/>
+                        <field name="related_product_id" domain="[('product_tmpl_id', '=', product_tmpl_id)]" readonly="False" attrs="{'invisible': [('product_tmpl_id', '=', False)]}" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."/>
+                        <field name="product_id" attrs="{'invisible': [('product_tmpl_id', '!=', False)]}" readonly="context.get('set_product_readonly', False)" force_save="1" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."/>
```
Comment (svs-odoo) [inline addons/stock/views/stock_production_lot_views.xml]: ... and undo this change.

File: addons/stock/models/stock_production_lot.py [hunk 12]
```diff
@@ -14,9 +14,13 @@ class ProductionLot(models.Model):
         'Lot/Serial Number', default=lambda self: self.env['ir.sequence'].next_by_code('stock.lot.serial'),
         required=True, help="Unique Lot/Serial Number")
     ref = fields.Char('Internal Reference', help="Internal reference number in case it differs from the manufacturer's lot/serial number")
+    product_tmpl_id = fields.Many2one('product.template', string='Product Template')
     product_id = fields.Many2one(
         'product.product', 'Product',
         domain=[('type', '=', 'product')], required=True)
+    related_product_id = fields.Many2one(
+        'product.product', 'Lot Product',
+        domain=[('type', '=', 'product')], related="product_id")
```
[the remark this answers is not in the document]
Comment (svs-odoo) [reply addons/stock/models/stock_production_lot.py]: The method:
``` python3
def _domain_product_id(self):
    domain = [('type', '=', 'product')]
    if self.env.context.get('default_product_tmpl_id'):
        domain = expression.AND([domain, [('product_tmpl_id', '=', self.env.context['default_product_tmpl_id'])]])
    return domain
```

The product field:
``` python3
product_id = fields.Many2one(
    'product.product', 'Product',
    domain=lambda self: self._domain_product_id(), required=True)
```

File: addons/stock/views/stock_production_lot_views.xml (hunk 11 above)
Comment (svs-odoo) [reply addons/stock/views/stock_production_lot_views.xml]: More precisely, undo this change except the change about the readonly on `product_id` (`readonly="context.get('set_product_readonly', False)" force_save="1"`).

File: addons/mrp/views/mrp_bom_views.xml [hunk 13]
```diff
@@ -100,7 +100,6 @@
                         <page string="Miscellaneous">
                             <group>
                                 <group>
-                                    <field name="sequence"/>
```
Comment (svs-odoo) [inline addons/mrp/views/mrp_bom_views.xml]: Don't forget to update the commit message as you only hide it now.

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by sle-odoo** [comment]
robodoo r-

**Comment by sle-odoo** [comment]
robodoo r-

**Comment by sle-odoo** [comment]
robodoo r+ rebase-merge
